### ISSUE TITLE
Can't access nested map fields

### ISSUE THREAD

**Issue by shira-redlich** [zhouzhuojie/conditions#16]
I'm trying to use nested args. e.g.:
"{
		"EventData" : {
			"Count" : 2,
			"sum" : 3,
		},
	}
"
and my expression is:
`({EventData}{Count} > 3)`

But I'm getting error:
`argument: EventData.Count not found`

Seems like there is an issue in https://github.com/zhouzhuojie/conditions/blob/master/evaluator.go#L68
We try to access the argument directly instead of checking if this is a nested arg

**Comment by zhouzhuojie** [comment]
Thanks for reporting this — the behavior you hit is expected with the **composed key** syntax `{EventData}{Count}`, not a missing nested-map feature overall.

### What’s going on

- `{EventData}{Count}` is parsed as a **single flat key**: `EventData.Count` → lookup is `args["EventData.Count"]`, not `args["EventData"]["Count"]`.
- Your payload has a **nested** map under `EventData`, so that flat key is not present → `argument: EventData.Count not found`.

Nested JSON-style data **is** supported via **path** syntax inside one brace group:

```go
expr, _ := conditions.Parse(`({EventData.Count} > 3)`)
args := map[string]interface{}{
    "EventData": map[string]interface{}{"Count": 2, "sum": 3},
}
ok, err := conditions.Evaluate(expr, args) // ok == false, err == nil
```

### Two supported styles (see README)

| Syntax | Args shape | Example |
|--------|------------|---------|
| Composed `{a}{b}` | Flat dotted keys | `{"EventData.Count": 2}` |
| Path `{a.b}` | Nested maps (e.g. after `json.Unmarshal`) | `{"EventData": {"Count": 2}}` |

Tests explicitly document that composed keys do **not** walk nested maps: use `{EventData.Count}` (or flatten keys) for your shape.

If you’d like `{EventData}{Count}` to also try nested traversal when the flat key is missing, that would be a small API extension — happy to discuss in a follow-up PR. For now, `{EventData.Count}` should unblock you.
